### PR TITLE
Rename authentication for authorization

### PR DESCRIPTION
**Pull request by mdelolmo**
> Authentication is the process of ascertaining that somebody really is who they claim to be.
> 
> Authorization refers to rules that determine who is allowed to do what. E.g. Adam may be authorized to create and delete databases, while Usama is only authorised to read.

The main purpose of the library is to authorize a third party app, not to authenticate it. Authentication takes place elsewhere.

### CODE REVIEW

**Comment by mdelolmo** [comment]
@veronikapetruskova

File: auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java [hunk 1]
```diff
@@ -219,22 +219,22 @@
...
-     * {@link com.spotify.sdk.android.authentication.AuthenticationClient.AuthenticationClientListener}
+     * {@link AuthorizationClient.AuthenticationClientListener}
      */
-    private AuthenticationHandler mCurrentHandler;
+    private AuthorizationHandler mCurrentHandler;
 
-    private List<AuthenticationHandler> mAuthenticationHandlers = new ArrayList<>();
+    private List<AuthorizationHandler> mAuthorizationHandlers = new ArrayList<>();
 
     private AuthenticationClientListener mAuthenticationClientListener;
```
Comment (veronikapetruskova) [inline auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java]: should be `AuthorizationClientListener`

File: auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java [hunk 2]
```diff
@@ -219,22 +219,22 @@
...
 
     /**
-     * The activity that receives and processes the result of authentication flow
+     * The activity that receives and processes the result of authorization flow
      * and returns it to the context activity that invoked the flow.
-     * An instance of {@link com.spotify.sdk.android.authentication.LoginActivity}
+     * An instance of {@link LoginActivity}
      */
     private final Activity mLoginActivity;
     private boolean mAuthenticationPending;
```
Comment (veronikapetruskova) [inline auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java]: `mAuthorizationPending`

File: auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java [hunk 3]
```diff
@@ -486,26 +486,26 @@ private void sendComplete(AuthenticationHandler authHandler, AuthenticationRespo
         }
     }
 
-    private boolean tryAuthenticationHandler(final AuthenticationHandler authHandler, AuthenticationRequest request) {
-        authHandler.setOnCompleteListener(new AuthenticationHandler.OnCompleteListener() {
+    private boolean tryAuthenticationHandler(final AuthorizationHandler authHandler, AuthorizationRequest request) {
```
Comment (veronikapetruskova) [inline auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java]: `tryAuthorizationHandler`

File: auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java [hunk 4]
```diff
@@ -520,7 +520,7 @@ public void onError(Throwable error) {
         return true;
     }
 
-    private void closeAuthenticationHandler(AuthenticationHandler authHandler) {
+    private void closeAuthenticationHandler(AuthorizationHandler authHandler) {
```
Comment (veronikapetruskova) [inline auth-lib/src/main/java/com/spotify/sdk/android/auth/AuthorizationClient.java]: `closeAuthorizationHandler `

File: auth-lib/src/main/java/com/spotify/sdk/android/auth/LoginActivity.java [hunk 5]
```diff
@@ -149,11 +149,11 @@ protected void onDestroy() {
     protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
         super.onActivityResult(requestCode, resultCode, intent);
         if (requestCode == REQUEST_CODE) {
-            AuthenticationResponse.Builder response = new AuthenticationResponse.Builder();
+            AuthorizationResponse.Builder response = new AuthorizationResponse.Builder();
 
             if (resultCode == RESULT_ERROR) {
                 Log.d(TAG, "Error authenticating");
```
Comment (veronikapetruskova) [inline auth-lib/src/main/java/com/spotify/sdk/android/auth/LoginActivity.java]: `Error authorizing`

File: auth-lib/src/test/java/com/spotify/sdk/android/auth/AuthorizationClientTest.java [hunk 6]
```diff
@@ -122,36 +122,36 @@ public void getResponseShouldReturnAParcableIfEverythingIsOk() {
...
-                = mock(AuthenticationClient.AuthenticationClientListener.class);
+        AuthorizationClient.AuthenticationClientListener mockListener
+                = mock(AuthorizationClient.AuthenticationClientListener.class);
         client.setOnCompleteListener(mockListener);
 
-        AuthenticationResponse response = new AuthenticationResponse.Builder()
-                .setType(AuthenticationResponse.Type.ERROR)
+        AuthorizationResponse response = new AuthorizationResponse.Builder()
+                .setType(AuthorizationResponse.Type.ERROR)
                 .setError("Authentication failed")
```
Comment (veronikapetruskova) [inline auth-lib/src/test/java/com/spotify/sdk/android/auth/AuthorizationClientTest.java]: `Authorization failed`

File: auth-sample/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -25,7 +25,7 @@
     android:id="@+id/activity_main"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context="com.spotify.sdk.android.authentication.sample.MainActivity">
+    tools:context="com.spotify.sdk.android.authentication.sample.com.spotify.sdk.android.auth.sample.MainActivity">
```
Comment (veronikapetruskova) [inline auth-sample/src/main/res/layout/activity_main.xml]: is this correct?

Comment (mdelolmo) [reply auth-sample/src/main/res/layout/activity_main.xml]: Nope... Good catch

**Comment by mdelolmo** [comment]
@veronikapetruskova Good catches. I addressed them

**Pull request merged**